documents/invoice/model.go

```diff
@@ -259,79 +264,53 @@ func (i *Invoice) getInvoiceSalts(invoiceData *invoicepb.InvoiceData) (*proofs.S
 	return i.InvoiceSalts, nil
 }
 
-// ID returns document identifier.
-// Note: this is not a unique identifier for each version of the document.
-func (i *Invoice) ID() ([]byte, error) {
-	coreDocModel, err := i.PackCoreDocument()
-	if err != nil {
-		return []byte{}, err
-	}
-	return coreDocModel.Document.DocumentIdentifier, nil
-}
-
 // PackCoreDocument packs the Invoice into a Core Document
 // If the, Invoice is new, it creates a valid identifiers
-func (i *Invoice) PackCoreDocument() (*documents.CoreDocumentModel, error) {
-	invoiceData := i.createP2PProtobuf()
-	serializedInvoice, err := proto.Marshal(invoiceData)
+func (i *Invoice) PackCoreDocument() (cd coredocumentpb.CoreDocument, err error) {
+	invData := i.createP2PProtobuf()
+	data, err := proto.Marshal(invData)
 	if err != nil {
-		return nil, errors.NewTypedError(err, errors.New("couldn't serialise InvoiceData"))
+		return cd, errors.New("couldn't serialise InvoiceData: %v", err)
 	}
 
-	invoiceAny := any.Any{
+	embedData := &any.Any{
 		TypeUrl: documenttypes.InvoiceDataTypeUrl,
```

Comment:
You could use the method you added below here:
```suggestion
		TypeUrl: i.DocumentType()
```